@xen-orchestra/web-core/lib/components/ui/checkbox/UiCheckbox.vue

```diff
@@ -0,0 +1,429 @@
+  align-items: center;
+  gap: 0.8rem;
+
+  .fake-checkbox {
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+    height: 1.6rem;
+    width: 1.6rem;
+    transition:
+      background-color 0.125s ease-in-out,
+      border-color 0.125s ease-in-out;
+    border: 0.2rem solid transparent;
+    border-radius: 0.2rem;
+    box-shadow: var(--shadow-100);
+
+    .icon {
+      font-size: 0.75rem;
+      position: absolute;
+      color: var(--color-normal-txt-item);
+      transition: opacity 0.125s ease-in-out;
+
+      filter: drop-shadow(0 0.0625em 0.5em rgba(0, 0, 0, 0.1)) drop-shadow(0 0.1875em 0.1875em rgba(0, 0, 0, 0.06))
+        drop-shadow(0 0.1875em 0.25em rgba(0, 0, 0, 0.08));
+    }
+  }
+
+  .ui-checkbox {
+    font-size: inherit;
+    position: absolute;
+    pointer-events: none;
+    opacity: 0;
+
+    &.indeterminate + .fake-checkbox > .icon {
+      opacity: 1;
+      color: var(--color-normal-txt-item);
+    }
+
+    & + .fake-checkbox > .icon {
+      opacity: 0;
+    }
+
+    &:checked + .fake-checkbox > .icon {
+      opacity: 1;
+    }
+
+    &:disabled {
+      &.accent--brand {
+        & + .fake-checkbox {
+          cursor: not-allowed;
+          border-color: var(--color-neutral-txt-secondary);
+
+          > .icon {
+            color: var(--color-neutral-txt-secondary);
+          }
+        }
+
+        &:checked + .fake-checkbox,
+        &.indeterminate + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-normal-item-disabled);
+        }
+      }
+
+      &.accent--success {
+        & + .fake-checkbox {
+          cursor: not-allowed;
+          border-color: var(--color-neutral-txt-secondary);
+
+          > .icon {
+            color: var(--color-neutral-txt-secondary);
+          }
+        }
+
+        &:checked + .fake-checkbox,
+        &.indeterminate + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-success-item-disabled);
+        }
+      }
+
+      &.accent--warning {
+        & + .fake-checkbox {
+          cursor: not-allowed;
+          border-color: var(--color-neutral-txt-secondary);
+
+          > .icon {
+            color: var(--color-neutral-txt-secondary);
+          }
+        }
+
+        &:checked + .fake-checkbox,
+        &.indeterminate + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-warning-item-disabled);
+        }
+      }
+
+      &.accent--danger {
+        & + .fake-checkbox {
+          cursor: not-allowed;
+          border-color: var(--color-neutral-txt-secondary);
+
+          > .icon {
+            color: var(--color-neutral-txt-secondary);
+          }
+        }
+
+        &:checked + .fake-checkbox,
+        &.indeterminate + .fake-checkbox {
+          border-color: transparent;
+          background-color: var(--color-danger-item-disabled);
+        }
+      }
+    }
+
+    &:not(:disabled) {
```

Comment:
What do you think of writing this block above the `:disabled` one? It would avoid a nested level while not changing the behavior of the CSS.
Also, I think the `:disabled` states can be nested within each variant group, like this:
```css
&.accent--brand {
      & + .fake-checkbox {
        border-color: var(--color-normal-item-base);
      }

      /* other styles... */ 

      &:active + .fake-checkbox {
        border-color: var(--color-normal-item-active);
      }

      &:checked + .fake-checkbox {
        border-color: transparent;
        background-color: var(--color-normal-item-base);
      }

      &:checked:hover + .fake-checkbox {
        background-color: var(--color-normal-item-hover);
      }

      &:checked:active + .fake-checkbox {
        background-color: var(--color-normal-item-active);
      }

      &:disabled {
        & + .fake-checkbox {
          cursor: not-allowed;
          border-color: var(--color-neutral-txt-secondary);

          > .icon {
            color: var(--color-neutral-txt-secondary);
          }
        }

        &:checked + .fake-checkbox,
        &.indeterminate + .fake-checkbox {
          border-color: transparent;
          background-color: var(--color-normal-item-disabled);
        }
      }
    }
```